Cannot deploy the new OpenZeppelin account contract

Tried using `starknet-rs` to deploy the latest [account contract from OpenZeppelin](https://github.com/OpenZeppelin/cairo-contracts/blob/5731e7134e755ad9b821963bf7d20f06acd1d355/openzeppelin/account/Account.cairo).

The transaction is [not working](https://alpha4.starknet.io/feeder_gateway/get_transaction?transactionHash=0x0232db7959b3ccc22b6e1321fc412025a7d9733b90db809c29e47601b5ef5e3f):

```json
{
  "status": "REJECTED",
  "transaction_failure_reason": {
    "tx_id": 1256114,
    "code": "TRANSACTION_FAILED",
    "error_message": "Error at pc=0:45:\nGot an exception while executing a hint.\nCairo traceback (most recent call last):\nUnknown location (pc=0:509)\nUnknown location (pc=0:499)\nUnknown location (pc=0:280)\nUnknown location (pc=0:242)\nUnknown location (pc=0:219)\n\nTraceback (most recent call last):\n  File \"<hint6>\", line 5, in <module>\nAssertionError: normalize_address() cannot be used with the current constants."
  },
  "transaction": {
    "contract_address": "0x615161661cdbea1c112383ce703d33299bef0e796903b88ec5601010ee22091",
    "contract_address_salt": "0x6c8035ea0cc6d4b0d933969db1367d7380adcaa462184b814f2def2bd9adf5",
    "constructor_calldata": [
      "1"
    ],
    "transaction_hash": "0x232db7959b3ccc22b6e1321fc412025a7d9733b90db809c29e47601b5ef5e3f",
    "type": "DEPLOY"
  }
}
```

--- a/starknet-core/Cargo.toml
+++ b/starknet-core/Cargo.toml
@@ -19,7 +19,7 @@ base64 = "0.13.0"
 ethereum-types = "0.12.1"
 hex = "0.4.3"
 serde = { version = "1.0.133", features = ["derive"] }
-serde_json = "1.0.74"
+serde_json = { version = "1.0.74", features = ["arbitrary_precision"] }
 serde_with = "1.12.0"
 sha3 = "0.10.0"
 thiserror = "1.0.30"

--- a/starknet-core/src/types/contract_code.rs
+++ b/starknet-core/src/types/contract_code.rs
@@ -1,6 +1,6 @@
 use super::{super::serde::unsigned_field_element::UfeHex, FieldElement};
 
-use serde::{Deserialize, Serialize};
+use serde::{de::Error as DeError, Deserialize, Serialize};
 use serde_with::serde_as;
 
 #[serde_as]
@@ -11,7 +11,7 @@ pub struct ContractCode {
     pub abi: Option<Vec<AbiEntry>>,
 }
 
-#[derive(Debug, Clone, Serialize, Deserialize)]
+#[derive(Debug, Clone, Serialize)]
 #[serde(tag = "type", rename_all = "lowercase")]
 pub enum AbiEntry {
     Constructor(Constructor),
@@ -82,6 +82,47 @@ pub struct Member {
     pub name: String,
     pub offset: u64,
     pub r#type: String,
+}
+
+// We need to manually implement this because `arbitrary_precision` doesn't work with `tag`:
+//   https://github.com/serde-rs/serde/issues/1183
+impl<'de> Deserialize<'de> for AbiEntry {
+    fn deserialize<D>(deserializer: D) -> Result<Self, D::Error>
+    where
+        D: serde::Deserializer<'de>,
+    {
+        let temp_value = serde_json::Value::deserialize(deserializer)?;
+        match &temp_value["type"] {
+            serde_json::Value::String(type_str) => match &type_str[..] {
+                "constructor" => Ok(AbiEntry::Constructor(
+                    Constructor::deserialize(temp_value).map_err(|err| {
+                        DeError::custom(format!("invalid constructor variant: {}", err))
+                    })?,
+                )),
+                "function" => Ok(AbiEntry::Function(
+                    Function::deserialize(temp_value).map_err(|err| {
+                        DeError::custom(format!("invalid function variant: {}", err))
+                    })?,
+                )),
+                "struct" => Ok(AbiEntry::Struct(Struct::deserialize(temp_value).map_err(
+                    |err| DeError::custom(format!("invalid struct variant: {}", err)),
+                )?)),
+                "l1_handler" => Ok(AbiEntry::L1Handler(
+                    L1Handler::deserialize(temp_value).map_err(|err| {
+                        DeError::custom(format!("invalid l1_handler variant: {}", err))
+                    })?,
+                )),
+                "event" => Ok(AbiEntry::Event(Event::deserialize(temp_value).map_err(
+                    |err| DeError::custom(format!("invalid event variant: {}", err)),
+                )?)),
+                _ => Err(DeError::custom(format!(
+                    "unknown ABI entry type: {}",
+                    type_str
+                ))),
+            },
+            _ => Err(DeError::custom("invalid type field")),
+        }
+    }
 }
 
 #[cfg(test)]

